calypso/csadmin/commands.go

```diff
@@ -0,0 +1,198 @@
+package main
+
+import (
+	"go.dedis.ch/cothority/v3/calypso/csadmin/clicontracts"
+	"gopkg.in/urfave/cli.v1"
+)
+
+var cmds = cli.Commands{
+	{
+		Name:      "authorize",
+		Usage:     "store the byzcoin-id that should be trusted to create new LTS",
+		Aliases:   []string{"a"},
+		ArgsUsage: "private.toml",
+		Action:    authorize,
+	},
+	{
+		Name:  "dkg",
+		Usage: "handles DKG operations",
+		Subcommands: cli.Commands{
+			{
+				Name:   "start",
+				Usage:  "starts a DKG given the instance ID of an LTS",
+				Action: dkgStart,
+				Flags: []cli.Flag{
+					cli.StringFlag{
+						Name:   "bc",
+						EnvVar: "BC",
+						Usage:  "the ByzCoin config to use (required)",
+					},
+					cli.StringFlag{
+						Name:  "instid, i",
+						Usage: "the instance id of the spawned LTS contract",
+					},
+					cli.BoolFlag{
+						Name:  "export, x",
+						Usage: "the public key is exported to stdout",
+					},
+				},
+			},
+		},
+	},
+	{
+		Name:   "decrypt",
+		Usage:  "decrypts a symmetric key given the proofs of write and read instances",
+		Action: decrypt,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:   "bc",
+				EnvVar: "BC",
+				Usage:  "the ByzCoin config to use (required)",
+			},
+			cli.StringFlag{
+				Name:  "writeid, w",
+				Usage: "instance id of the write instance",
+			},
+			cli.StringFlag{
+				Name:  "readid, r",
+				Usage: "instance id of the read instance",
+			},
+			cli.BoolFlag{
+				Name:  "export, x",
+				Usage: "the DecryptReply is exported to stdout",
+			},
+		},
+	},
+	{
+		Name:   "recover",
+		Usage:  "recover an encrypted key given in STDIN the DecryptKeyReply struct",
+		Action: recover,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:   "bc",
+				EnvVar: "BC",
+				Usage:  "the ByzCoin config to use (required)",
+			},
+			cli.StringFlag{
+				Name:  "key",
+				Usage: "path to the private.toml file (default is admin key)",
+			},
+			cli.BoolFlag{
+				Name:  "export, x",
+				Usage: "the decrypted data is exported to stdout",
+			},
+		},
+	},
+	{
+		Name:  "contract",
+		Usage: "use contracts",
```

Comment:
this usage information is not too useful, maybe briefly describe which contracts are available?

Reply:
Text is updated. I prefer not mentioning the list of contracts since it is likely to evolve and I will forget to update it. This information is dynamically available with `-h`.